### DISCUSSION TITLE
9.5 very slow for Sonoff

### DISCUSSION THREAD

**Discussion by neutmute** [arendst/Tasmota#12884] [Q&A] [answered]
I just moved house and ordered a FAN03 and TX wallplate - brand new. I also have 2 x Sonoff SV, TH16 and a Sonoff Basic from the old house - so I have just flashed numerous devices from Sonoff default firmware and upgraded from older sonoff (v6 through to v8) on others.

Every single device runs dead slow on v9.5 - taking 20-30 seconds for a new Web UI page to be served. Sometimes it gets faster a little while after startup. I can click Toggle and it won't switch the relay for a minute or so.

I took one of the SV's and upgraded it incrementally from 6.5 to each major version and found 9.1 to be the last version that has a usable Web UI (as far as response times go). After flashing each device down to 9.1, it is back to normal, snappy behaviour.

I don't see any other complaints for 9.5 which is really odd when the Web UI is effectively unusable for 100% of my v9.5 devices until downgraded. Perhaps it is something to do with my network setup? Any suggestions?

**Comment by crllsn** [comment]
Had same problem. Try this.
Sleep 0
MqttTimeout 15
MqttWifiTimeout 5000
SetOption41 105
ntpserver1 <put a local ntp server if you do not have internet access>

ntp server was a major culprit since I'm in a new house with no internet access, then the router. With this setup everything is smooth.

**Reply by sfromis** [reply]
Tasmota is not designed to operate without Wifi and NTP access.

**Reply by neutmute** [reply]
ah! I did set the NtpServer but I suspect my VLAN/firewall is blocking the request because the time never updated.
thanks for the tip - will investigate and try and fix the NTP routing.

**Reply by neutmute** [reply]
@sfromis  - wifi I understand but I don't understand why NTP is so important? 
I orchestrate via OpenHab so AFAIK correct local time on the tasmota shouldn't matter?

**Reply by sfromis** [reply]
Tasmota is not fully initialized without a local time, as features of Tasmota directly rely on having such. You can set the local time by other means than NTP if needed.

**Reply by barbudor** [reply]
NTP doesn't necessarily mean an internet access.
You may already have a NAS or server that runs 24/7 and that can act as a NTP server.

**Comment by arendst** [comment, marked as answer]
It's often not even the NTP server entry but it's the lack of DNS server connection. A DNS lookup takes up to 10 seconds until it timesout. This is hardcoded in the core and can be changed for an ESP8266 but is fixed in an ESP32 (even 14 seconds).

So to get a faster response from the NTP service change a NTP server name to it's IP address bypassing the DNS lookup. Latest development versions do more relaxed NTP lookup and will further enhance user experience response.

**Reply by digiblur** [reply]
So weird to me that people block DNS, NTP, etc from Tasmota devices and yet many of them let other massive projects run full reign on the network.

**Reply by sfromis** [reply]
Yeah, for me the real reason for even setting up a local NTP server is that I do not like having Tasmota devices hammer public NTP servers over 10000 times a week. Security-wise, Tasmota devices are something of the least concerning. Of course, I do not let them be publicly available servers. And in case of my network getting breached, I'd have much worse things to worry about than Tasmota.

BTW. I do use IP address instead of local DNS lookup for the NTP server, and most of my firmware upgrade happens via a local caching server. Thus it would have no "cost" blocking them from Internet access, but I see no real benefit, either.

**Reply by digiblur** [reply]
Absolutely, many routers and such have NTP on them available.  I point to a couple local NTP IPs as well.   I was just speaking on a larger scale as I've seen it many times.  So much time and effort on massive blocking of any traffic to Tasmota yet they let something with hundreds of packages and much more horsepower like Home Assistant run free and do whatever.  Or even worse some closed source devices as well.

**Comment by kaweksl** [comment]
Had same issue here, I got separate VLAN for IoT devices. all traffic is denied by default and allowed only Home Assistant to/from device and DHCP. Devices can't even talk to each other (client to client forwarding disabled).

When DNS is not reachable webui is responding slowly. Looks like tasmota tries to resolve ntp server names.
To solve that issue i had to allow traffic to local NTP server and set NTPServer to its ip address. No access to DNS needed.

PS: DNS may be used to get internet access, search for "dns tunnelling"

**Comment by eturpin** [comment]
I had a similar issue with my iFan04-L. By watching `ping` output I could see that the device would freeze up for ~15 seconds, handle all responses, then freeze up again. I tried all versions of Tasmota on the "Download" page:
Current stable: v10.0.0
Development branch: 20211020
v9.3.1
v9.1.0
 v8.5.1
 v7.2.0
 v6.7.1

The ifan04-l would become permanently unresponsive after connecting to my wifi network on every version except v10.0.0 and the development branch. v10.0.0 and the dev branch "worked," but had the sleeping issue. Setting "Sleep 0" fixed the issue.
